examples/low-level-api/create_vc.rs

```diff
@@ -37,17 +42,43 @@ pub async fn create_vc() -> Result<()> {
 
   println!("Credential JSON > {:#}", credential);
 
-  // Validate the verifiable credential
+  // Before passing this credential on the issuer wants to validate that some properties
+  // of the credential satisfy their expectations.
+  // In order to validate a credential the issuer's DID Document needs to be resolved.
+  // Since the issuer wants to issue and verify several credentials without publishing updates to their DID Document
+  // the issuer decides to resolve their DID Document up front now so they can re-use it.
+
   let resolver: Resolver = Resolver::new().await?;
-  let validation: CredentialValidation = common::check_credential(&resolver, &credential).await?;
-  println!("Credential Validation > {:#?}", validation);
-  assert!(validation.verified);
+  let resolved_issuer_doc: ResolvedIotaDocument = resolver.resolve(issuer_doc.id()).await?;
+  // Validate the credential's signature using the issuer's DID Document, the credential's semantic structure,
+  // that the issuance date is not in the future and that the expiration date is not in the past:
+  CredentialValidator::validate(
+    &credential,
+    &CredentialValidationOptions::default(),
+    &resolved_issuer_doc,
+    FailFast::Yes,
+  )?;
+  // The issuer is now sure that the credential they are about to issue satisfies their expectations
+  // hence the credential is now serialized to JSON before passing it to the subject in a secure manner.
+  // This means that the credential is NOT published to the tangle where it can be accessed by anyone.
+  let credential_json: String = credential.to_json()?;
 
-  Ok(())
+  Ok(credential_json)
 }
 
 #[tokio::main]
 async fn main() -> Result<()> {
-  let _ = create_vc().await?;
+  // Obtain a JSON representation of a credential issued to us
+  let credential_json: String = create_vc().await?;
+  // Run some basic validations on this credential so we can immediately contact the issuer if something is not right
+  let credential: Credential = Credential::from_json(credential_json.as_str())?;
+  // We can use the Resolver to conveniently validate basic properties of the credential.
+  let resolver: Resolver = Resolver::new().await?;
+  // We use FailFast::No so all errors get accumulated if there are any. Then we can later inform the issuer about
+  // everything that is wrong with the credential they issued to us and they will hopefully send us a new correct
+  // credential
+  resolver
+    .verify_credential(&credential, &CredentialValidationOptions::default(), FailFast::No)
+    .await?;
   Ok(())
```

Comment:
Good points. With regards to this: 
> but it may be misleading since one typically only validates a Credential after issuance when included in a Presentation. We would want to discourage developers from thinking that a Credential alone is a sufficient proof of claims, as it lacks the checks in Presentation such as proof-of-holder, challenge etc.

I am wondering whether the `verify_credential` convenience method is somewhat incompatible with that sentiment. Maybe the resolver should only have a convenience method for presentation validation/verification and developers who need to work with the credentials directly should use the `CredentialValidator`. What do you think?

Reply:
I also considered that. However, you showed the valid use-case of verifying credentials one issues or receives in this example, so it's a bit tricky whether or not we want to support that.

We should get other perspectives and opinions but I think we should remove `Resolver::verify_credential` entirely and switch to using `IotaDocument` (or even a generic `Document` but that depends on how future methods/sub-methods are to be implemented) in the validators to avoid needing to resolve for low-level checks, but not in `Resolver`.